File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/PdfDocument.java
class PdfDocument extends Document {
    public PdfDocument(String name) {
        super(name);
    }
    @Override
    void open() {
        System.out.println("Opening PDF document: " + name);
    }
    @Override
    void save() {
        System.out.println("Saving PDF " + name);
    }
}

File: Week-1_Design Patterns and Principles/2_FactoryMethodPatternExample/Code/PdfDocument.cs
public class PdfDocument : Document
{
    public PdfDocument(string name) : base(name) {}
    public override void Open()
    {
        Console.WriteLine("Opening PDF document: " + name);
    }
    public override void Save()
    {
        Console.WriteLine("Saving PDF " + name);
    }
}
